internal/tools/cloudflare/dns_test.go

```diff
@@ -0,0 +1,506 @@
+			expected: nil,
+		},
+		{
+			description:     "failure with empty zone name",
+			customerDNSName: "",
+			zoneNameList:    []string{},
+			dnsEndpoints:    []string{"load.balancer.endpoint"},
+			setupName: func(zoneNameList []string, customerDNSName string) (zoneName string, found bool) {
+				return "", false
+			},
+			setupID: func(zoneName string) (zoneID string, err error) {
+				return "", nil
+			},
+			setupDNS: func(ctx context.Context, zoneID string, rr cf.DNSRecord) (*cf.DNSRecordResponse, error) {
+				return &cf.DNSRecordResponse{}, nil
+			},
+			expected: errors.New("hosted zone for \"\" domain name not found"),
+		},
+		{
+			description:     "failure with no DNS endpoints",
+			customerDNSName: "customer.cloud.mattermost.com",
+			zoneNameList:    []string{"cloud.mattermost.com"},
+			dnsEndpoints:    []string{},
+			setupName: func(zoneNameList []string, customerDNSName string) (zoneName string, found bool) {
+				return "", false
+			},
+			setupID: func(zoneName string) (zoneID string, err error) {
+				return "", nil
+			},
+			setupDNS: func(ctx context.Context, zoneID string, rr cf.DNSRecord) (*cf.DNSRecordResponse, error) {
+				return &cf.DNSRecordResponse{}, nil
+			},
+			expected: errors.New("no DNS endpoints provided for Cloudflare creation request"),
+		},
+		{
+			description:     "failure with empty string DNS endpoint",
+			customerDNSName: "customer.cloud.mattermost.com",
+			zoneNameList:    []string{"cloud.mattermost.com"},
+			dnsEndpoints:    []string{""},
+			setupName: func(zoneNameList []string, customerDNSName string) (zoneName string, found bool) {
+				return "", false
+			},
+			setupID: func(zoneName string) (zoneID string, err error) {
+				return "", nil
+			},
+			setupDNS: func(ctx context.Context, zoneID string, rr cf.DNSRecord) (*cf.DNSRecordResponse, error) {
+				return &cf.DNSRecordResponse{}, nil
+			},
+			expected: errors.New("DNS endpoint was an empty string"),
+		},
+		{
+			description:     "failure with zone ID fetching",
+			customerDNSName: "customer.cloud.mattermost.com",
+			zoneNameList:    []string{"cloud.mattermost.com"},
+			dnsEndpoints:    []string{"load.balancer.endpoint"},
+			setupName: func(zoneNameList []string, customerDNSName string) (zoneName string, found bool) {
+				return "", false
+			},
+			setupID: func(zoneName string) (zoneID string, err error) {
+				return "", nil
+			},
+			setupDNS: func(ctx context.Context, zoneID string, rr cf.DNSRecord) (*cf.DNSRecordResponse, error) {
+				return &cf.DNSRecordResponse{}, nil
+			},
+			expected: errors.New("failed to fetch Zone ID from Cloudflare"),
+		},
+	}
+
+	for _, s := range samples {
+		t.Run(s.description, func(t *testing.T) {
+			mockCF.mockGetZoneName = s.setupName
+			mockCF.mockGetZoneID = s.setupID
+			mockCF.mockCreateDNSRecord = s.setupDNS
+			client := NewClientWithToken(mockCF)
+			err := client.CreateDNSRecord(s.customerDNSName, s.zoneNameList, s.dnsEndpoints, logger)
+			if err != nil {
```

Comment:
```suggestion
			if s.expected != nil {
```
Perhaps we could even skip the if completely and use just:
```go
assert.Equal(t, s.expected, err)
```
This should check if errors are the same but also pass if both sides are `nil`.

Reply:
It was failing with the `Equal` the `EqualError` I think it works fine